pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -41,13 +43,15 @@ func NewReconcileClusterDeployment(
 	kubeClient kubernetes.Interface,
 	informerHolder *source.InformerHolder,
 	recorder events.Recorder,
+	mcRecorder kevents.EventRecorder,
 ) *ReconcileClusterDeployment {
 
 	return &ReconcileClusterDeployment{
 		client:         client,
 		kubeClient:     kubeClient,
 		informerHolder: informerHolder,
 		recorder:       recorder,
+		mcRecorder:     mcRecorder,
```

Comment:
why do we still need the original recorder?

Reply:
yes, because we use the openshift library go to apply some resources, and the library go needs the openshift event recorder.